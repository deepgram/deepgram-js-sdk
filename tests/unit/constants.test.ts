import {
  DEFAULT_HEADERS,
  DEFAULT_URL,
  DEFAULT_OPTIONS,
  SOCKET_STATES,
  CONNECTION_STATE,
} from "../../src/lib/constants";

describe("Unit Tests - Constants and Configuration", () => {
  describe("DEFAULT_HEADERS", () => {
    it("should have correct structure and dynamic values", () => {
      const headers = { ...DEFAULT_HEADERS };

      // Replace dynamic values for consistent snapshots
      const result = {
        ...headers,
        "X-Client-Info": headers["X-Client-Info"].replace(/v[\d.]+/, "v[VERSION]"),
        "User-Agent": headers["User-Agent"]
          .replace(/\/[\d.]+/, "/[VERSION]")
          .replace(/(node|bun)\/[\d.]+/, "$1/[VERSION]"),
      };

      expect(result).toMatchSnapshot();
    });
  });

  describe("DEFAULT_URL and DEFAULT_OPTIONS", () => {
    it("should have correct default configuration", () => {
      const config = {
        DEFAULT_URL,
        // Simplified to avoid TypeScript header indexing issues
        DEFAULT_OPTIONS_STRUCTURE: {
          hasGlobal: !!DEFAULT_OPTIONS.global,
          hasAgent: !!DEFAULT_OPTIONS.agent,
          globalHasFetch: !!DEFAULT_OPTIONS.global?.fetch,
          globalHasWebsocket: !!DEFAULT_OPTIONS.global?.websocket,
          agentHasFetch: !!DEFAULT_OPTIONS.agent?.fetch,
          agentHasWebsocket: !!DEFAULT_OPTIONS.agent?.websocket,
        },
      };

      expect(config).toMatchSnapshot();
    });
  });

  describe("Enums", () => {
    it("should have correct enum values", () => {
      const enums = {
        SOCKET_STATES,
        CONNECTION_STATE,
      };

      expect(enums).toMatchSnapshot();
    });
  });
});
