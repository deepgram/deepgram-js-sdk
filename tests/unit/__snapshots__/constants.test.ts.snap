// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests - Constants and Configuration DEFAULT_HEADERS should have correct structure and dynamic values 1`] = `
{
  "Content-Type": "application/json",
  "User-Agent": "@deepgram/sdk/[VERSION]-automated node/[VERSION]",
  "X-Client-Info": "@deepgram/sdk; server; v[VERSION]-automated",
}
`;

exports[`Unit Tests - Constants and Configuration DEFAULT_URL and DEFAULT_OPTIONS should have correct default configuration 1`] = `
{
  "DEFAULT_OPTIONS_STRUCTURE": {
    "agentHasFetch": true,
    "agentHasWebsocket": true,
    "globalHasFetch": true,
    "globalHasWebsocket": true,
    "hasAgent": true,
    "hasGlobal": true,
  },
  "DEFAULT_URL": "https://api.deepgram.com",
}
`;

exports[`Unit Tests - Constants and Configuration Enums should have correct enum values 1`] = `
{
  "CONNECTION_STATE": {
    "Closed": "closed",
    "Closing": "closing",
    "Connecting": "connecting",
    "Open": "open",
  },
  "SOCKET_STATES": {
    "0": "connecting",
    "1": "open",
    "2": "closing",
    "3": "closed",
    "closed": 3,
    "closing": 2,
    "connecting": 0,
    "open": 1,
  },
}
`;
